### PR TITLE
Fixed bin/setup to work in correct order

### PR DESCRIPTION
**Pull request by vshatravenko**

### CODE REVIEW

**Comment by mod** [comment]
@vshatravenko we need instead to generate a seed file in the workbench

File: db/seeds.rb [hunk 1]
```diff
@@ -13,11 +13,18 @@
...
 if File.exist?([Rails.root, '/config', '/seed.yml'].join)
-  seed_file = File.join(Rails.root, 'config', 'seed.yml')
-  seed = YAML.load(File.open(seed_file))
+  seed_file   = File.join(Rails.root, 'config', 'seed.yml')
+  seed        = YAML.load(File.open(seed_file))
+
+  app_seed    = seed['oauth_application']
+  application = Doorkeeper::Application.new(skipauth: app_seed['skipauth'], name: app_seed['oauth_app_name'], redirect_uri: app_seed['oauth_callback_url'])
+
+  unless ENV['WORKBENCH'].nil?
```
Comment (dkkoval) [inline db/seeds.rb]: Let's do it like this: `if ENV['WORKBENCH'].in?(%w[ 1 true ])`. Also we don't need `seed.yml` in this case so check if it's the workbench before.

File: bin/setup [hunk 2]
```diff
@@ -19,10 +19,13 @@ chdir APP_ROOT do
...
 
-  puts "\n=== Preparing assets ==="
-  system! 'bin/rake yarn:install assets:precompile'
-
   puts "\n== Preparing database =="
   system! 'cp config/database.yml.example config/database.yml'
   system! 'bin/rails db:create db:migrate'
+
+  puts "\n== Seeding database =="
+  system! 'bin/rails db:seed'
```
Comment (dkkoval) [inline bin/setup]: Let's move it to the end since it outputs some credentials and we're lazy to scroll up 😄

File: config/seed.yml [hunk 3]
```diff
@@ -1,3 +1,4 @@
 oauth_application:
-  oauth_app_name:     'Client App'
-  oauth_callback_url: 'http://localhost:3001/oauth/callback'
\ No newline at end of file
+  oauth_app_name:     'Local Peatio'
```
Comment (dkkoval) [inline config/seed.yml]: Maybe simply `name`?

File: config/seed.yml [hunk 4]
```diff
@@ -1,3 +1,4 @@
...
+  oauth_callback_url: 'http://localhost:8000/auth/barong/callback'
```
Comment (dkkoval) [inline config/seed.yml]: and `callback_url`

File: db/seeds.rb (hunk 1 above)
Comment (mod) [reply db/seeds.rb]: this need to move in yml files seed.yml that will be mounted.

File: db/seeds.rb [hunk 5]
```diff
@@ -6,18 +6,31 @@
...
+
+  puts 'Admin credentials: %s' % [admin.password]
 
+  # Create applications from seed
+  seed['applications'].each do |app|
+    unless app['uid'].nil? && app['secret'].nil?
+      result = Doorkeeper::Application.new(skipauth: app['skipauth'], name: app['name'], redirect_uri: app['callback_url'], uid: app['uid'], secret: app['secret'])
+    else
+      result = Doorkeeper::Application.new(skipauth: app['skipauth'], name: app['name'], redirect_uri: app['callback_url'])
+    end
```
Comment (mod) [inline db/seeds.rb]: why you don't simply run
Doorkeeper::Application.new(app)

Comment (vshatravenko) [reply db/seeds.rb]: true :sweat_smile:

**Pull request merged**